sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/CloudEvent.cs

```diff
@@ -298,30 +304,48 @@ private async Task<T> GetDataInternal<T>(ObjectSerializer serializer, bool async
         /// Deserialized payload of the event. Returns null if there is no event data.
         /// Returns <see cref="BinaryData"/> for unknown event types.
         /// </returns>
-        public object GetData()
+        public BinaryData GetData() =>
+            GetDataInternal();
+
+        /// <summary>
+        /// Deserializes the event payload into a system event type or
+        /// returns the payload of the event wrapped as <see cref="BinaryData"/>. Using BinaryData,
+        /// one can deserialize the payload into rich data, or access the raw JSON data using <see cref="BinaryData.ToString()"/>.
+        /// </summary>
+        /// <returns>
+        /// Deserialized payload of the event. Returns null if there is no event data.
+        /// Returns <see cref="BinaryData"/> for unknown event types.
+        /// </returns>
+        public Task<BinaryData> GetDataAsync() =>
```

Comment:
Adding an Async method here based on API Review feedback that suggested doing this.